### PR TITLE
Do not skip TAG characters in glyph substitution

### PR DESCRIPTION
**Pull request by khaledhosny**
Hide them like Mongolian Free Variation Selectors instead.

Fixes https://github.com/behdad/harfbuzz/issues/463

### CODE REVIEW

File: src/hb-ot-layout-private.hh [hunk 1]
```diff
@@ -274,6 +275,10 @@ _hb_glyph_info_set_unicode_props (hb_glyph_info_t *info, hb_buffer_t *buffer)
        * https://github.com/behdad/harfbuzz/issues/234
        */
       if (unlikely (hb_in_range (u, 0x180Bu, 0x180Du))) props |= UPROPS_MASK_FVS;
+      /* TAG cgaracters used in emoji flags:
```
Comment (jfkthame) [inline src/hb-ot-layout-private.hh]: typo, "cgaracters"

File: src/hb-ot-layout-private.hh [hunk 2]
```diff
@@ -375,7 +380,7 @@ _hb_glyph_info_is_default_ignorable (const hb_glyph_info_t *info)
 static inline hb_bool_t
 _hb_glyph_info_is_default_ignorable_and_not_fvs (const hb_glyph_info_t *info)
```
Comment (jfkthame) [inline src/hb-ot-layout-private.hh]: Should we update the name of this function to add "_or_tag"? Or maybe use an alternative such as _hb_glyph_info_is_ignorable_for_skipping_iterator?

File: src/hb-ot-layout-private.hh [hunk 3]
```diff
@@ -241,6 +241,7 @@ enum hb_unicode_props_flags_t {
   UPROPS_MASK_GEN_CAT	= 0x001Fu,
   UPROPS_MASK_IGNORABLE	= 0x0020u,
   UPROPS_MASK_FVS	= 0x0040u, /* MONGOLIAN FREE VARIATION SELECTOR 1..3 */
+  UPROPS_MASK_TAG       = 0x0080u, /* TAG characters */
```
Comment (jfkthame) [inline src/hb-ot-layout-private.hh]: AFAICS, this doesn't really need to be a separate bit, we could just rename FVS to FVS_OR_TAG. I guess it's up to Behdad whether he prefers to keep them separate for clarity, or merge to leave the other bit free for some future use.

Comment (behdad) [inline src/hb-ot-layout-private.hh]: Yeah, I was thinking of changing the FVS to something like HIDDEN instead.  Let's keep the 0x0080 flag for future.

Comment (khaledhosny) [inline src/hb-ot-layout-private.hh]: I was split about this, couldn’t come up with a good name 😄. Reused and renamed the FVS flag.

File: src/hb-ot-layout-private.hh (hunk 1 above)
Comment (khaledhosny) [inline src/hb-ot-layout-private.hh]: Fixed.

**Comment by behdad** [comment]
Thanks!

**Comment by khaledhosny** [comment]
Should I merge this now?

**Comment by behdad** [comment]
> Should I merge this now?

I thought I did.  Doing now.

**Pull request merged**